app/Pain/Pain/Interface/Camera/RectangleDetectionViewController.swift

```diff
@@ -168,19 +165,69 @@ class RectangleDetectionViewController: CameraViewController {
     }
     
     private func filterRectangles(rectangles: [Detection]) -> [Detection] {
-        var rectangles = filterRectanglesByBounds(rectangles: rectangles)
-        rectangles = filterRectanglesByInclusion(rectangles: rectangles)
+        let rectangles = filterRectanglesByBounds(rectangles: rectangles)
+//        rectangles = filterRectanglesByInclusion(rectangles: rectangles)
         return rectangles
     }
     
     private func filterRectanglesByBounds(rectangles: [Detection]) -> [Detection] {
-        return rectangles
+        let bounds = rootLayer.bounds
+        let top = coverWeights.top * bounds.height
+        let bottom = (1 - coverWeights.bottom) * bounds.height
+        let left = coverWeights.left * bounds.width
+        let right = (1 - coverWeights.right) * bounds.width
+        let ratio = CGFloat(1.0 / 14.0)
+        return rectangles.filter { detection in
+            let rectangle = detection.relocated
+            return rectangle.points.filter { p in
+                let closeHorizontal =
+                    (abs(p.y - top) < bounds.height * ratio || abs(p.y - bottom) < bounds.height * ratio)
+                    && left - bounds.width * ratio < p.x && p.x < right + bounds.width * ratio
+                let closeVertical =
+                    (abs(p.x - left) < bounds.width * ratio || abs(p.x - right) < bounds.width * ratio)
+                        && top - bounds.height * ratio < p.y && p.y < bottom + bounds.height * ratio
+                return closeHorizontal || closeVertical
+            }.count == rectangle.points.count
+        }
     }
     
+    // This is not used at the moment, but I will leave it here just in case
     private func filterRectanglesByInclusion(rectangles: [Detection]) -> [Detection] {
-        return rectangles
+        let crossProduct = { (o: CGPoint, a: CGPoint, b: CGPoint) -> Double in
+            Double((a.x - o.x) * (b.y - o.y) - (a.y - o.y) * (b.x - o.x))
+        }
+        return rectangles.filter { detection in
+            let rectangle = detection.relocated
+            var keep = true
+            for otherDetection in rectangles {
+                let otherRectangle = otherDetection.relocated
+                if rectangle == otherRectangle {
+                    continue
+                }
+                var rectangleOutside = false
+                for point1 in rectangle.points {
+                    var pointOutside = false
+                    for i in 0..<4 {
+                        print(crossProduct(otherRectangle.points[i], otherRectangle.points[(i + 1) % 4], point1))
```

Comment:
Do you need this print to stay here in production?

Reply:
No, I removed it.